test/src/org/omegat/core/segmentation/SRXTest.java

```diff
@@ -26,138 +27,178 @@
+        }
 
-    /**
-     * Test SRX#loadFromDir produce SRX object properly.
-     * <p>
-     * MapRule#getLanguageCode should return Language Code
-     * defined in LanguageCode class.
-     * MapRule#getLanguage should return a localized name of language.
-     * The test here check both values.
-     * OmegaT 6.0 and before,
-     */
-    @Test
-    public void testSrxReaderDefault() {
-        assertTrue(segmentDefault.exists());
-        SRX srx = SRX.loadFromDir(segmentDefault);
-        assertNotNull(srx);
-        assertTrue(srx.isCascade());
-        List<MapRule> mapRuleList = srx.getMappingRules();
-        assertNotNull(mapRuleList);
-        assertEquals(18, mapRuleList.size());
-        for (MapRule mapRule : mapRuleList) {
-            if (mapRule.getPattern().equals("JA.*")) {
-                assertEquals(LanguageCodes.JAPANESE_CODE, mapRule.getLanguage());
-                assertEquals(OStrings.getString(LanguageCodes.JAPANESE_KEY), mapRule.getLanguageName());
+        /**
+         * Test SRX#loadFromDir produce SRX object properly.
+         * <p>
+         * MapRule#getLanguageCode should return Language Code defined in
+         * LanguageCode class. MapRule#getLanguage should return a localized
+         * name of language. The test here check both values. OmegaT 6.0 and
+         * before,
+         */
+        @Test
+        public void testSrxReaderDefault() {
+            assertTrue(SEGMENT_DEFAULT.exists());
+            SRX srx = SRX.loadFromDir(SEGMENT_DEFAULT);
+            assertNotNull(srx);
+            assertTrue(srx.isCascade());
+            List<MapRule> mapRuleList = srx.getMappingRules();
+            assertNotNull(mapRuleList);
+            assertEquals(18, mapRuleList.size());
+            for (MapRule mapRule : mapRuleList) {
+                if (mapRule.getPattern().equals("JA.*")) {
+                    assertEquals(LanguageCodes.JAPANESE_CODE, mapRule.getLanguage());
+                    assertEquals(OStrings.getString(LanguageCodes.JAPANESE_KEY), mapRule.getLanguageName());
+                }
             }
+            assertEquals("2.0", srx.getVersion());
+            assertTrue(srx.isSegmentSubflows());
+        }
+    }
+
+    public static class SRXMigrateTest {
+
+        @org.junit.Rule
+        public final LocaleRule localeRule = new LocaleRule(new Locale("en"));
+
+        @org.junit.Rule
+        public final TemporaryFolder folder = TemporaryFolder.builder().assureDeletion().build();
+
+        @Test
+        public void testSrxMigration() throws IOException {
+            File segmentConf = Paths.get(SEGMENT_CONF_BASE, "locale_en", "segmentation.conf").toFile();
+            File configDir = folder.newFolder();
+            SRXTest.testSrxMigration(segmentConf, configDir);
+        }
+    }
+
+    public static class SRXMigrateJaTest {
+
+        @org.junit.Rule
+        public final LocaleRule localeRule = new LocaleRule(new Locale("ja"));
```

Comment:
This is a hack to change System Locale to run unit test cases.

Because `LocaleRule` utility class is not related to the OmegaT functions, please ignore it if you don't know.